### ISSUE TITLE
Unable to preview GCODE files

### ISSUE THREAD

**Issue by R4toLoko** [SoftFever/OrcaSlicer#712]
TLDR: If you need to load a GCODE file to check if you have the right profiles in place, that won't happen forcing you to slice everything again just to be sure.

OS is unable to load its own GCODE for preview:

![image](https://user-images.githubusercontent.com/35049512/230749905-31c35b4d-4136-4164-85a5-2092a0a54360.png)

But the file was sliced via OS and ironically, if I try to load it on PS, it loads, broken but loads.

![image](https://user-images.githubusercontent.com/35049512/230749931-9c54405a-3ea8-42b0-8cf3-eee067e4fd38.png)

This is a GCODE file sliced via PS.
This is what you expect to see with OS

![image](https://user-images.githubusercontent.com/35049512/230749991-4ef5ba03-9a3c-47fe-8b91-0603f08d5fa8.png)

Side note: OS does not have "Load GCODE" like PS but drag and release.

**Comment by julie777** [comment]
Gcode files should have variables defined (or placeholders as BL calls them) for the type of printer and other information (plate type, etc.) to indicate whether they are usable or not for a given printer.

**Comment by H4kunaM4tata** [comment]
> Gcode files should have variables defined (or placeholders as BL calls them) for the type of printer and other information (plate type, etc.) to indicate whether they are usable or not for a given printer.
> 

This is one of the most annoying problem, buy have to slice things over and over because it's unable to load a GCODE file like PrusaSlicer

**Comment by R4toLoko** [comment]
@SoftFever the problem has not been solved on Linux at least and 1.6.3 beta.

![Peek 2023-05-22 18-53](https://github.com/SoftFever/OrcaSlicer/assets/35049512/3bb450bb-8c70-42c4-8368-66452600c04e)

**Comment by SoftFever** [comment]
@R4toLoko 
Can you share your gcode?
Mine works fine.
[111.zip](https://github.com/SoftFever/OrcaSlicer/files/11537709/111.zip)

**Comment by R4toLoko** [comment]
@SoftFever I am so sorry for the delay on this.
I tried to load your gcode file and nope.
Let me ask others.

![image](https://github.com/SoftFever/OrcaSlicer/assets/35049512/deb23555-5e82-4b51-b257-9c76f4a0204f)

**Comment by R4toLoko** [comment]
This appears to be a Linux bug, Mac seems to be working just fine.
Confirmed with other users.

**Comment by R4toLoko** [comment]
This has been completed, tested and it is working. 
Thanks a lot.
